route/charity-router.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const {Router} = require('express');
+const Charity = require('../model/charity.js');
+const httpErrors = require('http-errors');
+const bearerAuth = require('../lib/bearer-auth-middleware.js');
+
+const charityRouter = module.exports = new Router();
+
+let fuzzy = (filterTerm) => new RegExp('.*' + filterTerm.toLowerCase().split('').join('.*') + '.*');
+
+charityRouter.get('/charities', bearerAuth, (req, res, next) => {
+  let {page='0'} = req.query;
+  delete req.query.page;
+  page = Number(page);
+  if(isNaN(page))
+    page=0;
+  page = page < 0 ? 0 : page;
+
+  // Fuzzy Search
```

Comment:
This Fuzzy query Engine is awesome! Great work..

Yall are the first class to have both pagination and fuzzy search combined :)!!!

Reply:
Nice!